### ISSUE TITLE
HTML attributes that aren't pre-escaped?

### ISSUE THREAD

**Issue by mdwagner** [luckyframework/lucky#1823]
When using [LuckyHTML](https://luckyframework.github.io/lucky/Lucky/HTMLBuilder.html), I've noticed you don't need to run [`HTML.escape`](https://crystal-lang.org/api/1.8.2/HTML.html#escape%28string%3AString%29%3AString-class-method), as it does this for you behind the scenes (#9). Normally, this would make sense to do on everything: both text and attributes.

However, if you're using something like [htmx](https://htmx.org/) for your client-server interactions, there are situations where the [attributes _shouldn't_ be pre-escaped](https://htmx.org/attributes/hx-headers/).

As an example, this
<img width="254" alt="Screen Shot 2023-08-28 at 4 24 35 PM" src="https://github.com/luckyframework/lucky/assets/13472976/0e3cff2d-1160-4cc5-8a8c-2e9bf0441169">
becomes this
<img width="386" alt="Screen Shot 2023-08-28 at 4 24 48 PM" src="https://github.com/luckyframework/lucky/assets/13472976/cdac13ca-64c8-4194-8ccc-0d89f1d0771b">
which is undesirable (and broken).

We could provide a new tag method that _will not_ escape attributes (you can already do unescaped text with the `raw` tag helper), but idk if anyone has any other ideas.

**Comment by jwoertink** [comment]
Yeah, that makes sense. I've been talking with @watzon on doing better integration with HTMX for Lucky. Another thing needed would be allowing Actions as attribute values.

**Comment by mdwagner** [comment]
https://github.com/guilleiguaran/rails-htmx looks like a decent start at what an integration could look like for Lucky.

**Comment by watzon** [comment]
I was actually looking at the Laravel HTMX library as inspiration, but that Rails one could be a good pattern too

**Comment by jwoertink** [comment]
also https://github.com/mauricius/laravel-htmx

**Comment by rmarronnier** [comment]
@mdwagner : Thanks for opening this issue, I'm bumping my head into this issue right now (trying to pass the `X-CSRF-Token` value along with the htmx non get/post request).

Are you sure your request has a `myHeader` key with a `My value`  value ? (Mine don't with the same "badly" escaped string)

More on topic, when using the  https://luckyframework.org/html tool to "translate" this `<div hx-get="/example" hx-headers='{"myHeader": "My Value"}'>Get Some HTML, Including A Custom Header in the Request</div>` to lucky html, I get invalid code : `div "Get Some HTML, Including A Custom Header in the Request", hx_get: "/example", hx_headers: "{"myHeader": "My Value"}"
`
`Error: unexpected token: "myHeader"`

If the tool is reliable, then it is a bug :-)

**Comment by mdwagner** [comment]
@rmarronnier 
Yeah, that does seem like a bug in the html builder, which we should on address on https://github.com/luckyframework/website.

But to your point, yes, I can confirm it does add the correct headers to my outgoing request (I put `hx-headers` on the `body` tag).

**Comment by rmarronnier** [comment]
> Yeah, that does seem like a bug in the html builder, which we should on address on https://github.com/luckyframework/website.

Good idea, I'll open an issue.

> But to your point, yes, I can confirm it does add the correct headers to my outgoing request (I put hx-headers on the body tag).

I'll try that.
